addon/modifiers/basic-dropdown-trigger.ts

```diff
@@ -0,0 +1,157 @@
+import Modifier from 'ember-modifier';
+import { action } from '@ember/object';
+import { registerDestructor } from '@ember/destroyable';
+import hasMoved from '../utils/has-moved';
+import { Dropdown } from '../components/basic-dropdown';
+
+interface NamedArgs {
+  dropdown: Dropdown;
+  eventType?: 'click' | 'mousedown';
+  stopPropagation?: boolean;
+  [named: string]: unknown;
+}
+type PositionalArgs = unknown[];
+interface Args {
+  positional: PositionalArgs;
+  named: NamedArgs;
+};
+
+export default class DropdownTriggerModifier extends Modifier<Args> {
+  didSetup = false;
+
+  triggerElement?: HTMLElement;
+
+  toggleIsBeingHandledByTouchEvents: boolean = false;
+  touchMoveEvent?: TouchEvent;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args)
+    registerDestructor(this, cleanup)
+  }
+
+  modify(element: HTMLElement, positional: PositionalArgs, named: NamedArgs): void {
+    if (!this.didSetup) {
+      this.setup(element)
+      this.didSetup = true
+    }
+    this.update(element, positional, named)
+  }
+
+  setup(element: HTMLElement) {
+    // Keep a reference to the element for cleanup
+    this.triggerElement = element
+
+    if(!element.getAttribute('role')) element.setAttribute('role', 'button');
+
+    element.addEventListener('click', this.handleMouseEvent);
+    element.addEventListener('mousedown', this.handleMouseEvent);
+    element.addEventListener('keydown', this.handleKeyDown);
+    element.addEventListener('touchstart', this.handleTouchStart);
+    element.addEventListener('touchend', this.handleTouchEnd);
+  }
+
+  update(element: HTMLElement, _positional: PositionalArgs, named: NamedArgs) {
+    const { dropdown } = named
+
+    element.setAttribute('data-ebd-id', `${dropdown?.uniqueId}-trigger`);
```

Comment:
The only thing that concerns me is that updates set all those attributes even if they haven't changed. 
Do you know if modifiers have a way of only updating what has changed? 

Probably not a big deal, it's going to be performant enough, but I wondered.

Reply:
We can set a tracked property on the class, let's say `uniqueId`, then only set those attrs if that id changed
`if (dropdown?.uniqueId !== this.uniqueId) this.setUniqueAttributes(dropdown?.uniqueId)`

I felt the simplicity was more important

Reply:
I'd say that you don't have to worry. The trigger is a single component, so performance is not so critical. I would be more concerned if this was ember-power-select and this was each of the possibly hundreds of options. Keep it this simple.